presto-main/src/main/java/com/facebook/presto/operator/scalar/DayOfMonthWithEnglishSuffixPrinter.java

```diff
@@ -0,0 +1,186 @@
+            FormatUtils.appendUnpaddedInteger(appendable, day);
+            appendable.append(getSuffix(day));
+        }
+        catch (RuntimeException e) {
+            appendable.append('\ufffd');
+        }
+    }
+
+    @Override
+    public void printTo(StringBuffer buf, long instant, Chronology chrono, int displayOffset, DateTimeZone displayZone, Locale locale)
+    {
+        try {
+            printTo(buf, instant, chrono);
+        }
+        catch (IOException e) {
+            // StringBuffer does not throw IOException
+        }
+    }
+
+    @Override
+    public void printTo(Writer out, long instant, Chronology chrono, int displayOffset, DateTimeZone displayZone, Locale locale)
+            throws IOException
+    {
+        printTo(out, instant, chrono);
+    }
+
+    private void printTo(Appendable appendable, ReadablePartial partial) throws IOException
+    {
+        if (partial.isSupported(iFieldType)) {
+            try {
+                int day = partial.get(iFieldType);
+                FormatUtils.appendUnpaddedInteger(appendable, day);
+                appendable.append(getSuffix(day));
+            }
+            catch (RuntimeException e) {
+                appendable.append('\ufffd');
+            }
+        }
+        else {
+            appendUnknownString(appendable, MIN_PRINTED_LENGTH);
+        }
+    }
+
+    @Override
+    public void printTo(StringBuffer buf, ReadablePartial partial, Locale locale)
+    {
+        try {
+            printTo(buf, partial);
+        }
+        catch (IOException ex) {
+            // StringBuffer does not throw IOException
+        }
+    }
+
+    @Override
+    public void printTo(Writer out, ReadablePartial partial, Locale locale)
+            throws IOException
+    {
+        printTo(out, partial);
+    }
+
+    static void appendUnknownString(Appendable appendable, int len) throws IOException
+    {
+        for (int i = len; --i >= 0; ) {
+            appendable.append('\ufffd');
+        }
+    }
+
+    @Override
+    public int estimateParsedLength()
+    {
+        return MAX_PRINTED_LENGTH;
+    }
+
+    @Override
+    public int parseInto(DateTimeParserBucket bucket, String text, int position)
+    {
+        int limit = Math.min(MAX_PRINTED_LENGTH, text.length() - position);
+
+        int length = 0;
+        while (length < limit) {
+            char c = text.charAt(position + length);
+            if (c < '0' || c > '9') {
+                break;
+            }
+            length++;
+        }
+
+        if (length == 0) {
+            return ~position;
+        }
+
+        int value;
+        int i = position;
+        try {
+            value = text.charAt(i++) - '0';
+        }
+        catch (StringIndexOutOfBoundsException e) {
+            return ~position;
+        }
+        position += length;
+        while (i < position) {
+            value = ((value << 3) + (value << 1)) + text.charAt(i++) - '0';
```

Comment:
parseInto here is referencing the logic of NumberFormatter.parseInto in Joda-time.
https://github.com/JodaOrg/joda-time/blob/909046aa9aab5fae75e2152fd6ded722a4d5aba2/src/main/java/org/joda/time/format/DateTimeFormatterBuilder.java#L1304